Support for analysis with control samples

athos opened the pull request:
This PR changes `genomon-api` so that it accepts requests for DNA/RNA analysis with control panel.

With this patch, the handler would accept requests like the following:

```
$ curl -H 'Content-Type: application/json' \
  -d@- http://localhost:8000/api/pipelines/dna/runs <<EOS
{"tumor":{"r1":"s3://my-bucket/samples/r1.fastq","r2":"s3://my-bucket/samples/r2.fastq"},
 "controlpanel":["s3://my-bucket/samples/control_sample1.bam","s3://my-bucket/samples/control_samples2.bam"]}
EOS
```

And then it would generate a config file for `genomon_pipeline_cloud` that looks like:

```ini
[fastq]
tumor,s3://my-bucket/samples/r1.fastq,s3://my-bucket/samples/r2.fastq

[bam_tofastq]
control_sample0,s3://my-bucket/samples/sample1.bam
control_sample1,s3://my-bucket/samples/sample2.bam

[controlpanel]
controlpanel,control_sample0,control_sample1

[mutation_call]
tumor,None,controlpanel

[sv_detection]
tumor,None,controlpanel

[qc]
tumor
```

I've already confirmed the RNA pipeline works fine with a control panel after applying the patch.
As for the DNA pipeline, on the other hand, I think I've done the necessary work as far as it relates to `genomon-api`, but since [`genomon_pipeline_cloud`](https://github.com/chrovis/genomon_pipeline_cloud) does not support analysis with control panel at this moment, it still needs some more work to get the whole system to work all right.

alumi commented on test/src/genomon_api/handler_test.clj:
```diff
@@ -224,6 +224,16 @@
+       :controlpanel ["control-sample1" "control-sample2"]}
+      {:tumor {:r1 "tumor-r1" :r2 "tumor-r1"}
+       :normal {:r1 "normal-r1" :r2 "normal-r1"}
+       :controlpanel ["control-sample" "control-sample2"]}))
```
Check that `GET /api/pipelines/dna/runs/:run-id` contains the control panel samples

alumi commented on src/genomon_api/handler/pipelines/dna.clj:
```diff
@@ -34,8 +34,9 @@
-                       :tumor {:r1 string?, :r2 string?}}},
+   :parameters {:body {:tumor {:r1 string?, :r2 string?},
+                       (ds/opt :normal) {:r1 string?, :r2 string?},
+                       (ds/opt :controlpanel) [string?]}},
```
It's a trivial point but I prefer using `control-panel`.

athos commented:
@alumi Hi, thank you for the comments!

> Would you update the SQL to store the control panel samples for better reproducibility?

Sure, I will.
Will it suffice to store the control panel samples in a single text field of a certain format? Or do we have to set up a naïve 1:N relationship between `dna-`/`rna-runs` and control panel samples?

alumi commented:
> Will it suffice to store the control panel samples in a single text field of a certain format?

I think it will :+1:
Currently, we have no plan to support any specialized operations on control panels.

athos commented:
@alumi Sorry for the late response! 🙇 

I've made some more changes according to your suggestions since when I commented previously:

- Renamed `controlpanel` to `control-panel` / `control_panel` (a11ddce)
- Added another column to save a control panel into the `dna-runs` / `rna-runs` tables (d0e9fbf)
- Added more tests to check if a response from the DNA/RNA handler contains the stored control panel if any (006adbe)

A control panel, which is a collection of control samples, is stored in a column as a JSON string, for the time being.

I've also added some additional changes not coming from the above comments:

- Dropped pesky nil padding (like `{:normal {:r1 nil :r2 nil}, :control-panel nil, ...}`), which was internally required but eventually exposed as part of responses (c104397)
- Removed the control panel from `mutation_call`'s config (048d23f)
